modules/options/src/main/scala/scala/build/options/ScalacOpt.scala

```diff
@@ -10,15 +10,16 @@ final case class ScalacOpt(value: String) {
   /** @return raw key for the option (only if the key can be shadowed from the CLI) */
   private[options] def shadowableKey: Option[String] = key match
     case Some(key)
-        if ScalacOpt.repeatingKeys.exists(rKey => rKey.startsWith(key + ":") || rKey == key) => None
+        if ScalacOpt.repeatingKeys.exists(rKey => rKey.contains(":") || rKey == key) => None
```

Comment:
There seems to be something in this logic that I don't get for some reason. Is `rKey.contains(":")` ever true? None of the keys listed in `ScalacOpt.repeatingKeys` seems to contain `":"`

Reply:
Och right, this should be `key.contains`

Reply:
ah, that's a legacy inconsistency... because the `:` used to be there.
Now it indeed will never come true. 
Lemme adjust it.

Reply:
Should be fixed

Reply:
...I ended up reverting this change, as it breaks options shading/overriding from command line.
We should address this properly elsewhere.